App.tsx

```diff
@@ -1,18 +1,231 @@
-import { StyleSheet, Text, View } from 'react-native';
+import { useState } from "react";
+import { Pressable, StyleSheet, Text, View } from "react-native";
 
 export default function App() {
+  const [display, setDisplay] = useState("0");
+  const [firstInput, setFirstInput] = useState<string | undefined>();
+  const [secondInput, setSecondInput] = useState<string | undefined>();
+  const [operator, setOperator] = useState<string | undefined>();
+
+  const clearAll = () => {
+    resetInputs();
+    setDisplay("0");
+  };
+
+  const resetInputs = () => {
+    setFirstInput(undefined);
+    setSecondInput(undefined);
+    setOperator(undefined);
+  };
+
+  const calculateResult = () => {
+    if (operator === "/" && secondInput === "0") {
+      // Set error on divide by zero
+      setDisplay("Error");
+      return;
+    }
+
+    // TODO: Add validation
+
+    switch (operator) {
+      case "+":
+        setDisplay(`${+firstInput + +secondInput}`);
+        break;
+      case "-":
+        setDisplay(`${+firstInput - +secondInput}`);
+        break;
+      case "*":
+        setDisplay(`${+firstInput * +secondInput}`);
+        break;
+      case "/":
+        setDisplay(`${+firstInput / +secondInput}`);
+        break;
+    }
+
+    resetInputs();
+  };
+
+  const handleInputPress = (value: string) => {
+    const isOperator = ["+", "-", "*", "/"].includes(value);
+
+    if (!isOperator && !operator && !secondInput) {
+      const next = !firstInput ? value : firstInput + value;
+      setDisplay(next);
+      setFirstInput(next);
+      return;
+    }
+
+    if (isOperator && firstInput) {
+      if (!secondInput) {
+        // Update the operator until a secondInput is entered
+        setOperator(value);
+      } else {
+        // Multiple calculations not supported
+        return;
+      }
+    }
+
+    if (isOperator && !secondInput) {
+      setOperator(value);
+      return;
+    }
+
+    // Finally, populate the secondInput
+    const next = !secondInput ? value : secondInput + value;
+    setDisplay(next);
+    setSecondInput(next);
+  };
+
   return (
     <View style={styles.container}>
-      <Text>Open up App.tsx to start working on your app!</Text>
+      <View style={styles.row}>
+        <Pressable
+          style={[styles.button, styles.funcButton, styles.oneColumn]}
+          onPress={clearAll}
+        >
+          <Text>AC</Text>
+        </Pressable>
+        <Text style={styles.display}>{display}</Text>
+      </View>
+
+      <View style={styles.row}>
+        <Pressable
+          style={[styles.button, styles.oneColumn]}
+          onPress={() => handleInputPress("7")}
+        >
+          <Text>7</Text>
+        </Pressable>
```

Comment:
Ideally these buttons could be a separate, reusable component with styling and text options as props. I found this approach to be the fastest way to get something working before coming back to refactor later.